scripts/apps/vocabularies/controllers/VocabularyEditController.ts

```diff
@@ -10,16 +12,51 @@ VocabularyEditController.$inject = [
     'vocabularies',
     'metadata',
     'cvSchema',
+    'relationsService',
 ];
 
+interface IScope extends IScopeConfigController {
+    idRegex: string;
+    vocabulary: IVocabulary;
+    selectionTypes: IVocabularySelectionTypes;
+    closeVocabulary: () => void;
+    updateVocabulary: (result: any) => void;
+    issues: Array<any> | null;
+    _errorUniqueness: boolean;
+    errorMessage: string;
+    save: () => void;
+    requireAllowedTypesSelection: () => void;
+    addItem: () => void;
+    cancel: () => void;
+    model: any;
+    schema: any;
+    schemaFields: Array<any>;
+    itemsValidation: { valid: boolean };
+    removeItem: (index: number) => void;
+    customFieldTypes: Array<{id: string, label: string}>;
+    setCustomFieldConfig: (config: any) => void;
+    editForm: any;
+}
+
 const idRegex = '^[a-zA-Z0-9-_]+$';
 
-export function VocabularyEditController($scope, notify, api, vocabularies, metadata, cvSchema) {
+export function VocabularyEditController(
+    $scope: IScope, notify, api, vocabularies, metadata, cvSchema, relationsService,
+) {
     var origVocabulary = _.cloneDeep($scope.vocabulary);
 
     $scope.idRegex = idRegex;
     $scope.selectionTypes = VOCABULARY_SELECTION_TYPES;
 
+    if ($scope.matchFieldTypeToTab?.('related-content-fields', $scope.vocabulary.field_type)) {
```

Comment:
`matchFieldTypeToTab` looks to be [defined unconditionally](https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/apps/vocabularies/controllers/VocabularyConfigController.ts#L86). Why are you checking if it's defined? Can it be undefined?

Reply:
a lot of tests were failing because they create the scope but don't provide this function (and they don't really need it)

Reply:
Can you pass a dummy function from tests instead? We need to have the right structure so testing is easy but we shouldn't be adding if statements to the components/pages specifically for tests.